e2e/libs/utils/common_utils.py

```diff
@@ -18,39 +18,50 @@
 RETRY_INTERVAL = 1
 RETRY_INTERVAL_LONG = 2
 
+def try_loading_config():
+    if os.getenv('LONGHORN_CLIENT_URL') or config_utils.Config.get("LONGHORN_CLIENT_URL"):
+        config.load_kube_config()
+    else:
+        config.load_incluster_config()
+
 def k8s_core_api():
     c = Configuration()
     c.assert_hostname = False
     Configuration.set_default(c)
-    config.load_incluster_config()
+    try_loading_config()
     core_api = client.CoreV1Api()
     return core_api
 
 def k8s_cr_api():
     c = Configuration()
     c.assert_hostname = False
     Configuration.set_default(c)
-    config.load_incluster_config()
+    try_loading_config()
     cr_api = client.CustomObjectsApi()
     return cr_api
 
 def get_longhorn_api_client():
-    for i in range(RETRY_EXEC_COUNTS):
-        try:
-            config.load_incluster_config()
-            ips = get_mgr_ips()
-
-            # check if longhorn manager port is open before calling get_client
-            for ip in ips:
-                sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-                mgr_port_open = sock.connect_ex((ip, 9500))
-
-                if mgr_port_open == 0:
-                    client = 'http://' + ip + PORT + '/' #get_client(ip + PORT)
-                    break
-            return client
-        except Exception:
-            time.sleep(RETRY_INTERVAL)
+    if os.getenv('LONGHORN_CLIENT_URL'):
+        return os.getenv('LONGHORN_CLIENT_URL')
+    elif config_utils.Config.get("LONGHORN_CLIENT_URL"):
```

Comment:
This could be refined later after checking if `settings.ini` is necessary or not. FYI